matchtree.go

```diff
@@ -889,12 +889,19 @@ func (d *indexData) newMatchTree(q query.Q) (matchTree, error) {
 		}
 
 		if substr, ok := subMT.(*substrMatchTree); ok {
+			// Temporary: We have a feature flag for lazy decoding. If
+			// runeDocSections is nil it means we need to lazily decode on request.
+			sections := d.runeDocSections
+			if sections == nil {
+				sections = unmarshalDocSections(d.runeDocSectionsRaw, nil)
+			}
```

Comment:
@keegancsmith sorry to necrobump this PR, but is it intentional that the lazily-unmarshaled value isn't retained in `d.runeDocSections`?  Right now, roughly [50% of the allocated bytes](https://console.cloud.google.com/profiler;timespan=30d/zoekt-webserver;type=HEAP_ALLOC/alloc_space?referrer=search&project=sourcegraph-dev) in production are from repeatedly unmarshalling here.

Reply:
Don't hate me, but I have this TODO from april

```
** TODO ship disabled lazy decoding to subset of prod cluster
  [2022-04-04 Mon 15:50]
[[file:~/org-files/journals/2022/2022-04-Apr.org::*feature flag lazy decoding][feature flag lazy decoding]]
```

I suppose the time is right to switch off the lazy decoding :P

Reply:
I've scheduled this TODO on monday, don't wanna ship it and then have alerts go off over the weekend.

Reply:
hahaha, nice. Sounds good. This didn't come out of any actual problems. Just me occasionally perusing the profiling data

Reply:
https://github.com/sourcegraph/deploy-sourcegraph-cloud/pull/17531